internal/runners/initialize/init.go

```diff
@@ -50,184 +47,138 @@ func New(prime primeable) *Initialize {
-func inferLanguage(config projectfile.ConfigGetter) string {
+func inferLanguage(config projectfile.ConfigGetter) (string, string, bool) {
 	defaultProjectDir := config.GetString(constants.GlobalDefaultPrefname)
 	if defaultProjectDir == "" {
-		return ""
+		return "", "", false
 	}
 	defaultProj, err := project.FromPath(defaultProjectDir)
 	if err != nil {
-		return ""
+		return "", "", false
 	}
 	commitID := defaultProj.CommitUUID()
 	if commitID == "" {
-		return ""
+		return "", "", false
 	}
 	lang, err := model.FetchLanguageForCommit(commitID)
 	if err != nil {
-		return ""
+		return "", "", false
 	}
-	return fmt.Sprintf("%s@%s", lang.Name, lang.Version)
+	return lang.Name, lang.Version, true
 }
 
-func sanitize(params *RunParams, config projectfile.ConfigGetter) error {
-	if params.Language == "" {
-		params.Language = inferLanguage(config)
+func (r *Initialize) Run(params *RunParams) error {
+	if err := params.Namespace.Validate(); err != nil {
+		return locale.WrapInputError(err, "init_invalid_namespace_err", "The provided namespace argument is invalid.")
 	}
 
-	if params.Language == "" {
-		// Manually check for language requirement, because we need to fallback on the --language flag to support editor.V0
-		return locale.NewInputError("err_init_no_language", "You need to supply the [NOTICE]language[/RESET] argument, run [ACTIONABLE]`state init --help`[/RESET] for more information.")
-	}
-	langParts := strings.Split(params.Language, "@")
-	if len(langParts) > 1 {
-		params.version = langParts[1]
-	}
+	logging.Debug("Init: %s/%s %v", params.Namespace.Owner, params.Namespace.Project, params.Private)
 
-	// Disambiguate "python", defaulting to "python3" if no version was given.
-	if strings.ToLower(langParts[0]) == language.Python3.Requirement() {
-		if params.version == "" || strings.HasPrefix(params.version, "3") {
-			langParts[0] = language.Python3.String()
-		} else if strings.HasPrefix(params.version, "2") {
-			langParts[0] = language.Python2.String()
+	path := params.Path
+	if path == "" {
+		var err error
+		path, err = osutils.Getwd()
+		if err != nil {
+			return locale.WrapInputError(err, "err_init_sanitize_path", "Could not prepare path: {{.V0}}", err.Error())
 		}
 	}
 
-	params.language = language.Supported{language.MakeByName(langParts[0])}
-	if !params.language.Recognized() {
-		return language.UnrecognizedLanguageError(
-			params.Language,
-			language.RecognizedSupportedsNames(),
-		)
+	if fileutils.FileExists(filepath.Join(path, constants.ConfigFileName)) {
+		return locale.NewInputError("err_projectfile_exists")
 	}
 
-	// Fail if target dir already has an activestate.yaml
-	if fileutils.FileExists(filepath.Join(params.Path, constants.ConfigFileName)) {
-		absPath, err := filepath.Abs(params.Path)
-		if err != nil {
-			return errs.Wrap(err, "IO failure")
-		}
-		return locale.NewInputError("err_init_file_exists", "", absPath)
+	if _, err := fileutils.PrepareDir(path); err != nil {
```

Comment:
```suggestion
	if err := fileutils.MkdirUnlessExists(path); err != nil {
```

If you're not using the return value then you just need `MkdirUnlessExists`.